tests/headers/template.hpp

```diff
@@ -1,11 +1,26 @@
+// bindgen-flags: -- -std=c++11
+//
 template<typename T, typename U> class Foo {
     T m_member;
     T* m_member_ptr;
     T m_member_arr[1];
 };
 
+template<typename T> class B {
+    T m_member;
+};
+
 void bar(Foo<int, int> foo);
 
+struct C {
+    B<unsigned int> mB;
+    B<const int*> mBConstPtr;
+    B<const int> mBConst;
+    B<volatile int> mBVolatile;
+    B<const bool> mBConstBool;
+    B<const char16_t> mBConstChar;
```

Comment:
It might probably better to have more, at least something like `unsigned long long1 which has whitespace unrelated to cv, as well as things like `long double` to cover non-interger types, and probably `const decltype(nullptr)` might be a good one to test as well.